src/internal/operations/test/restore_helper_test.go

```diff
@@ -0,0 +1,245 @@
+// the clients themselves.
+func prepNewTestRestoreOp(
+	t *testing.T,
+	ctx context.Context, //revive:disable-line:context-as-argument
+	backupStore storage.Storage,
+	backupID model.StableID,
+	bus events.Eventer,
+	ctr *count.Bus,
+	sel selectors.Selector,
+	opts control.Options,
+	restoreCfg control.RestoreConfig,
+) (
+	operations.RestoreOperation,
+	*restoreOpDependencies,
+) {
+	var (
+		rod = &restoreOpDependencies{
+			acct: tconfig.NewM365Account(t),
+			st:   backupStore,
+		}
+		k = kopia.NewConn(rod.st)
+	)
+
+	err := k.Connect(ctx, repository.Options{})
+	require.NoError(t, err, clues.ToCore(err))
+
+	// kopiaRef comes with a count of 1 and Wrapper bumps it again
+	// we're so safe to close here.
+	rod.closer = func() {
+		err := k.Close(ctx)
+		assert.NoErrorf(t, err, "k close: %+v", clues.ToCore(err))
+	}
+
+	rod.kw, err = kopia.NewWrapper(k)
+	if !assert.NoError(t, err, clues.ToCore(err)) {
+		return operations.RestoreOperation{}, nil
+	}
+
+	rod.kms, err = kopia.NewModelStore(k)
+	if !assert.NoError(t, err, clues.ToCore(err)) {
+		return operations.RestoreOperation{}, nil
+	}
+
+	rod.sw = store.NewKopiaStore(rod.kms)
+
+	connectorResource := resource.Users
+	if sel.Service == selectors.ServiceSharePoint {
+		connectorResource = resource.Sites
+	}
+
+	rod.ctrl, rod.sel = ControllerWithSelector(
+		t,
+		ctx,
+		rod.acct,
+		connectorResource,
+		sel,
+		nil,
+		rod.close)
+
+	ro := newTestRestoreOp(
+		t,
+		ctx,
+		rod,
+		backupID,
+		bus,
+		ctr,
+		opts,
+		restoreCfg)
+
+	rod.sss = streamstore.NewStreamer(
+		rod.kw,
+		rod.acct.ID(),
+		rod.sel.PathService())
+
+	return ro, rod
+}
+
+// newTestRestoreOp accepts the clients required to compose a restore operation, plus
+// any other metadata, and uses them to generate a new restore operation.  This
+// allows restore chains to utilize the same temp directory and configuration
+// details.
+func newTestRestoreOp(
+	t *testing.T,
+	ctx context.Context, //revive:disable-line:context-as-argument
+	rod *restoreOpDependencies,
+	backupID model.StableID,
+	bus events.Eventer,
+	ctr *count.Bus,
+	opts control.Options,
+	restoreCfg control.RestoreConfig,
+) operations.RestoreOperation {
+	rod.ctrl.IDNameLookup = idname.NewCache(map[string]string{rod.sel.ID(): rod.sel.Name()})
+
+	ro, err := operations.NewRestoreOperation(
+		ctx,
+		opts,
+		rod.kw,
+		rod.sw,
+		rod.ctrl,
+		rod.acct,
+		backupID,
+		rod.sel,
+		restoreCfg,
+		bus,
+		ctr)
+	if !assert.NoError(t, err, clues.ToCore(err)) {
+		rod.close(t, ctx)
+		t.FailNow()
+	}
+
+	return ro
+}
+
+func runAndCheckRestore(
+	t *testing.T,
+	ctx context.Context, //revive:disable-line:context-as-argument
+	ro *operations.RestoreOperation,
+	mb *evmock.Bus,
+	expectItemsWritten int, // can be -1 to skip the check
+	acceptNoData bool,
+) *details.Details {
+	deets, err := ro.Run(ctx)
+	require.NoError(t, err, clues.ToCore(err))
+	require.NotEmpty(t, ro.Results, "the restore had non-zero results")
```

Comment:
add a comment about why this check is alright. At first glance it seems like a restore that restores no data would fail this check